tests/test_ioc_system.py

```diff
@@ -306,13 +347,42 @@ async def test_bobfiles_created(mocked_panda_standard_responses):
     assert len(old_files) == len(new_files)
 
 
-async def test_create_bobfiles_fails_if_files_present(standard_responses, tmp_path):
-    response_handler = ResponseHandler(standard_responses)
-    mocked_client = MockedAsyncioClient(response_handler)
+async def test_create_bobfiles_fails_if_files_present(tmp_path, new_random_test_prefix):
     Path(tmp_path / "PCAP.bob").touch()
 
     with pytest.raises(FileExistsError):
-        await _create_softioc(mocked_client, TEST_PREFIX, tmp_path)
+        Pvi.set_screens_dir(tmp_path, False)
+        Pvi.create_pvi_records(new_random_test_prefix)
+
+
+async def test_create_bobfiles_deletes_existing_files_with_clear_bobfiles(
+    tmp_path,
+    new_random_test_prefix,
+    clear_records,
+):
+    generated_bobfile = Path(tmp_path / "TOP.bob")
+    non_generated_bobfile = Path(tmp_path / "Blahblah.bob")
+    non_bobfile = Path(tmp_path / "Blahblah.txt")
+
+    generated_bobfile.touch()
+    assert generated_bobfile.read_text() == ""
+    non_generated_bobfile.touch()
+    non_bobfile.touch()
+
+    Pvi.set_screens_dir(tmp_path, True)
+    Pvi.add_pvi_info(
+        new_random_test_prefix + ":PCAP:TRIG_EDGE",
+        PviGroup.PARAMETERS,
+        SignalX("TRIG_EDGE", "Falling"),
+    )
+    Pvi.create_pvi_records(new_random_test_prefix)
+
+    await asyncio.sleep(1)
```

Comment:
What's this sleep for? Isn't `create_pvi_records` entirely synchronous?

Reply:
Correct, my mistake